server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -59,6 +62,119 @@ public class IndexingStats implements Writeable, ToXContentFragment {
      */
     public static class Stats implements Writeable, ToXContentFragment {
 
+        /**
+         * Tracks item level rest category class codes during indexing
+         *
+         * @opensearch.internal
+         */
+        public static class DocStatusStats implements Writeable, ToXContentFragment {
+
+            final AtomicLong[] docStatusCounter;
+
+            public DocStatusStats() {
+                this(Stream.generate(AtomicLong::new).limit(5).toArray(AtomicLong[]::new));
+            }
+
+            protected DocStatusStats(final AtomicLong[] docStatusCounter) {
+                this.docStatusCounter = docStatusCounter;
+            }
+
+            public DocStatusStats(StreamInput in) throws IOException {
+                docStatusCounter = new AtomicLong[5];
+
+                for (int i = 0; i < docStatusCounter.length; ++i) {
+                    docStatusCounter[i] = new AtomicLong(in.readLong());
+                }
+            }
+
+            public void inc(final int familyCode) {
```

Comment:
Since this is a public method, can we take the RestStatus here directly, and calculate the status code family to allow for a safer abstraction?